Extension/src/LanguageServer/settings.ts

```diff
@@ -312,88 +309,199 @@ export class CppSettings extends Settings {
-    public get clangTidyChecksEnabled(): string[] | undefined { return super.Section.get<string[]>("codeAnalysis.clangTidy.checks.enabled"); }
-    public get clangTidyChecksDisabled(): string[] | undefined { return super.Section.get<string[]>("codeAnalysis.clangTidy.checks.disabled"); }
-    public get clangTidyCodeActionShowDisable(): boolean | undefined { return super.Section.get<boolean>("codeAnalysis.clangTidy.codeAction.showDisable"); }
-    public get clangTidyCodeActionShowClear(): string { return super.Section.get<string>("codeAnalysis.clangTidy.codeAction.showClear") ?? "AllAndAllType"; }
-    public get clangTidyCodeActionShowDocumentation(): boolean | undefined { return super.Section.get<boolean>("codeAnalysis.clangTidy.codeAction.showDocumentation"); }
-    public get clangTidyCodeActionFormatFixes(): boolean { return super.Section.get<boolean>("codeAnalysis.clangTidy.codeAction.formatFixes") ?? true; }
+    private getAsBoolean(settingName: string): boolean {
+        const value: any | undefined = super.Section.get(settingName);
+        if (isBoolean(value)) {
+            return value;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default;
+    }
+
+    private getAsString(settingName: string): string {
+        const value: any | undefined = super.Section.get(settingName);
+        if (this.isValidEnum(settingName, value)) {
+            return value.toLowerCase(); // Return enum values in lowercase.
+        } else if (isString(value)) {
+            return value;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default;
+    }
+
+    private getAsNumber(settingName: string): number {
+        const value: any | undefined | null = super.Section.get(settingName);
+        const setting = getRawSetting("C_Cpp." + settingName);
+
+        if (isNumber(value)) {
+            if (setting.minimum !== undefined && setting.maximum !== undefined) { 
+                if (value >= setting.minimum && value <= setting.maximum) { // Check if the value is within the valid range.
+                    return value;
+                }
+            } else {
+                return value;
+            }
+        }   
+
+        return setting.default;
+    }
+
+    // This helper function returns the value of a setting as an array of strings with proper type validation.
+    // Additionally, it checks for valid enum values if an array of enums is detected.
+    private getAsArrayOfStrings(settingName: string, allowUndefinedEnums: boolean = false): string[] {
+        const setting = getRawSetting("C_Cpp." + settingName);
+        const value: any | undefined | null = super.Section.get(settingName);
+        if (isArrayOfString(value)) {
+            if (setting.items.enum !== undefined) {
+                if (!value.every(x => this.validateEnum(setting.items.enum, x))) {
```

Comment:
If `allowUndefinedEnums` is true, you don't need to check the values of the enums at all.

```suggestion
            if (!allowUndefinedEnums && setting.items.enum !== undefined) {
```